server/server.go

```diff
@@ -1238,6 +1236,9 @@ func (server *ArgoCDServer) newHTTPServer(ctx context.Context, port int, grpcWeb
 	// Dex reverse proxy and client app and OAuth2 login/callback
 	server.registerDexHandlers(mux)
 
+	th := util_session.WithAuthMiddleware(server.DisableAuth, server.ssoClientApp, server.settings, server.sessionMgr, terminal)
+	mux.Handle("/terminal", th)
+
```

Comment:
I had to move these lines below `server.registerDexHandlers` to avoid `a.ssoClientApp` being `nil`. I hope that doesn't have other side-effects.

Reply:
Based on the implementation of registerDexHandlers, if SSO is not configured, ssoClientApp will be nil. The auth middleware should handle the nil use case. Make sure to add an additional unit test to validate that (TestWithAuthMiddlewareWhenSSONotConfigured)